### ISSUE TITLE
MediaPlayer destructor hangs if object destroyed too quickly.

### ISSUE THREAD

**Issue by jaynabonne** [alexa/avs-device-sdk#2037]
**IMPORTANT**: Before you create an issue, please take a look at our [Issue Reporting Guide](https://github.com/alexa/avs-device-sdk/wiki/Issue-Reporting-Guide).

## Briefly summarize your issue:
As part of my ramping up our Alexa client, I have been adding in functionality piece by piece. I got to the point of adding the (GStreamer) MediaPlayer, and I noticed the app then wouldn't shut down properly. It just hung. By adding some logging, I determined that if the worker thread hasn't started up and entered g_main_loop_run yet, then the destructor will try to shut down the loop and then join the thread. However, since the loop hasn't been entered yet, the g_main_loop_quit call does nothing. And then when the thread finally enters the loop, it doesn't know it needs to quit. Meanwhile, the destructor is waiting on the thread to exit, which it never does.

I have worked around this by adding a small sleep in between my starting up the SDK and shutting it down. Eventually, this won't be an issue, when our app is full featured.

## What is the expected behavior?
The MediaPlayer destructor should not hang.

## What behavior are you observing?
The MediaPlayer destructor hangs if the internal worker thread has not yet started up. 

## Provide the steps to reproduce the issue, if applicable:
Create a MediaPlayer and then release it immediately.

## Tell us about your environment:
This issue was found in the Windows/MSYS2 environment.

### What version of the AVS Device SDK are you using?
1.26.0

### Tell us what hardware you're using:
- [X] Desktop / Laptop
- [ ] Raspberry Pi
- [ ] Other - tell us more:

### Tell us about your OS (Type & version):
- [ ] Linux
- [ ] MacOS 
- [ ] Raspbian Stretch
- [ ] Raspbian Jessy
- [X ] Other - tell us more: Windows 10, using MSYS2

[Alexa MediaPlayer hang.txt](https://github.com/alexa/avs-device-sdk/files/8813612/Alexa.MediaPlayer.hang.txt)

**Comment by jaynabonne** [comment]
In the attached log, logging prefixed with "++++++" is in the MediaPlayer destructor. Logging prefixed with "------" is in workerLoop. It more or less parallels the code, so I'm hoping it won't be too hard to see what's what.

**Comment by blazusiak-amazon** [comment]
Hello @jaynabonne,
Are you still experiencing this issue? If so, could you please provide more information to the question asked above?

**Comment by jaynabonne** [comment]
Hi.
Apologies for not getting back to you on this. I'm not experiencing this issue any longer, as our code has advanced to the point where this can no longer happen. I imagine it would still happen if we shut down quickly enough, but it's not a problem affecting our production code. 

In terms of threading, we're only using a single thread. However, the AVS code is being loaded as a .so dynamically. I'm not sure if that could be impacting things. If so, I would consider it an extremely minor edge case. (It probably is anyway.)

Thanks for following up with this. :)

**Comment by kjkh** [comment]
In that case, closing the issue. If further support is required, please create a new issue with a reference to this one.
